src/Route/RoutePageGenerator.php

```diff
@@ -89,7 +89,22 @@ public function update(SiteInterface $site, OutputInterface $output = null, $cle
 
         // no root url for the given website, create one
         if (!$root) {
-            $root = $this->pageManager->create([
+            // first search for a route
+            foreach ($this->router->getRouteCollection()->all() as $name => $route) {
+                if ('/' === $route->getPath()) {
+                    $requirements = $route->getRequirements();
+                    $name = trim($name);
+                    $root = $this->pageManager->create([
+                        'routeName' => $name,
+                        'name' => $name,
+                        'url' => $route->getPath(),
+                        'site' => $site,
+                        'requestMethod' => isset($requirements['_method']) ? $requirements['_method'] : 'GET|POST|HEAD|DELETE|PUT',
+                        'slug' => '/',
+                    ]);
+                }
+            }
+            $root = ($root) ? $root : $this->pageManager->create([
```

Comment:
Why the parenthesis? Also `$root ?: $this->pageManager->create([…` should be equivalent, right?